core/src/main/java/org/apache/struts2/interceptor/FetchMetadataInterceptor.java

```diff
@@ -73,15 +76,31 @@ public String intercept(ActionInvocation invocation) throws Exception {
             return invocation.invoke();
         }
 
-        logger.atDebug().log(
-            "Fetch metadata rejected cross-origin request to %s",
-            contextPath
-        );
+        LOG.debug("Fetch metadata rejected cross-origin request to [{}]", contextPath);
```

Comment:
To improve, I think `.warn` is better.

Reply:
Hi.  It looks like this interceptor could produce a lot of logging output, to the point that it might flood the application's logs.
Maybe the debug-level output is OK, since the developer can usually find the related forbidden (403) response in the access logs of the application server anyway ?
Another option could be to use a `.info` level for the message instead (could still be suppressed if developers set minimum log level to warning, but easier to differentiate than debug-level).  What do you think ?

Reply:
Hi,

I think you guys know better what log level is more appropriate for Struts devs. What I can tell you is that this code is not expected to reject many requests under normal circumstances, as every time this happens an endpoint is being fetched cross-origin, which indicates a potential CSRF attack more often than not and developers are expected to add endpoints that are exposed cross-origin to the exemption list. 

That is to say: we wouldn't really expect this log statement to be hit to the point of flooding the logs.